pkg/graph/schemas/schema.graphql

```diff
@@ -38,7 +38,25 @@ input TestRunFilter {
 }
 
 type Query {
-  testRuns: [TestRun!]!
+  testRuns(first: Int, after: String): TestRunConnection!
   testRun(testRunFilter: TestRunFilter!): [TestRun!]!
   testRunById(id: Int!): TestRun
 }
+
+type PageInfo {
+  hasNextPage: Boolean!
+  hasPreviousPage: Boolean!
+  startCursor: String!
```

Comment:
Why would the object coming from the api need to know about cursor position? Shouldn't that only be needed by the ui, and could it differ based on client/browser?

Reply:
Hi @jjchambl, if the client doesnt know about the cursor position, it will be unable to get the next page of testRuns. It would be needed by non-ui clients too, as the alternative(without pagination) is to return all testRuns(status-quo) which could be quite data intensive for the backend. No, it should not differ based on the client/browser as we are making clients use the pagination to get their data. 

e.g.
When querying for testRuns, you might start with a query like this:
```

query {
  testRuns(first: 10) {
    edges {
      cursor
      testRun {
        id
        testProjectName
      }
    }
    pageInfo {
      hasNextPage
      endCursor
    }
  }
}
```

If hasNextPage is true, you can fetch the next set of results using the endCursor as the after parameter:

`query {
  testRuns(first: 10, after: "cursor_value") {
    edges {
      cursor
      testRun {
        id
        testProjectName
      }
    }
    pageInfo {
      hasNextPage
      endCursor
    }
  }
}`